### ISSUE TITLE
Make action compatible with custom plugin system

### ISSUE THREAD

**Issue by 0xalpharush** [crytic/slither-action#16]
Slither has a custom plugin system and it'd be nice to have it easily integrate into the github action by registering a file(s) and including those detectors in the run. 
https://github.com/crytic/slither/tree/master/plugin_example

**Comment by elopez** [comment]
I was thinking about a way to implement this. What about a way to provide a custom requirements file, to be installed after Slither? I left an untested draft in #21 if you want to see how it would look like.

```yaml
name: Slither Analysis
on: [push]
jobs:
  analyze:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      # let's say plugins are bundled in the repository or checked out @ 'plugins/'
      - uses: crytic/slither-action@dev-slither-plugins
        with:
          target: 'src/'
          slither-plugins: 'plugins/requirements.txt'
```

**Comment by 0xalpharush** [comment]
Would the plugin have to be available on pip? Usually, it would just be a single python file with a custom detector.

**Comment by elopez** [comment]
Not necessarily, but it would have to be installable with pip (e.g. having the setup.py like in the linked example). Other than that, the requirements file could point to a local wheel, a folder or separate checked out repository, a (public) git or zip URL or so on.

**Comment by elopez** [comment]
Hey @fextr ! Feedback from interested parties would be great. Is the draft on #21 something that would be useful to you? Why or why not?

**Comment by elopez** [comment]
@fextr I've merged the proposed changes with a bit more of documentation to the dev branch, you should be able to try it out if you want. Just write a pip requirements file specifying what to install (e.g. just a txt file saying `slitherin` if you want to install the version from pip) and then provide the path to the file on the `slither-plugins` option. Use `crytic/slither-action@dev` so that Github picks up the dev branch.

```
      - uses: crytic/slither-action@dev
        with:
          ...
          slither-plugins: ./requirements-plugins.txt
          ...
```

**Comment by elopez** [comment]
@fextr released in v0.4.0 👍
